integrations/vault/secrets/pki/src/main/java/io/helidon/integrations/vault/secrets/pki/PkiSecretsImpl.java

```diff
@@ -16,58 +16,149 @@
+                                                                  VaultOptionalResponse.<byte[], byte[]>vaultResponseBuilder()
+                                                                          .entityProcessor(Function.identity()));
+        return CaCertificateGet.Response.builder()
+                .entity(response.entity().orElseThrow(() ->
+                                                              VaultRestException.builder()
+                                                                      .headers(response.headers())
+                                                                      .requestId(response.requestId())
+                                                                      .status(response.status())
+                                                                      .vaultErrors(response.errors())
+                                                                      .message(
+                                                                              "CA Certificate is expected to be present, but is"
+                                                                                      + " not available")
+                                                                      .build()))
+                .headers(response.headers())
+                .requestId(response.requestId())
+                .status(response.status())
+                .build();
     }
 
     @Override
     public VaultOptionalResponse<CertificateGet.Response> certificate(CertificateGet.Request request) {
-        return delegate.certificate(request).await();
+        String apiPath = mount + "/cert/" + request.serialNumber();
+
+        PkiFormat format = request.format();
+        if (format != PkiFormat.PEM) {
+            throw new UnsupportedOperationException("Only PEM encoded format is supported");
+        }
+
+        return restApi.get(apiPath, request, VaultOptionalResponse.<CertificateGet.Response, JsonObject>vaultResponseBuilder()
+                .entityProcessor(CertificateGet.Response::create));
     }
 
     @Override
     public CrlGet.Response crl(CrlGet.Request request) {
-        return delegate.crl(request).await();
+        PkiFormat format = request.format();
+
+        String apiPath = switch (format) {
+            case DER -> mount + "/crl";
+            case PEM, PEM_BUNDLE -> mount + "/crl/pem";
+        };
+
+        VaultOptionalResponse<byte[]> response = restApi.getBytes(apiPath,
+                                                                  request,
+                                                                  VaultOptionalResponse.<byte[], byte[]>vaultResponseBuilder()
+                                                                          .entityProcessor(Function.identity()));
+        return CrlGet.Response.builder()
+                .entity(response.entity().orElseThrow(() ->
+                                                              VaultRestException.builder()
```

Comment:
Same comment as above (excessive indentation)

Reply:
This is how the idea template is updating it so will touch base on this.

Reply:
```java
return CrlGet.Response.builder()
        .entity(response.entity()
                        .orElseThrow(() -> VaultRestException.builder()
                                .headers(response.headers())
                                .requestId(response.requestId())
                                .status(response.status())
                                .vaultErrors(response.errors())
                                .message(
                                        "CRL is expected to be present, but is not "
                                                + "available")
                                .build()))
        .headers(response.headers())
        .requestId(response.requestId())
        .status(response.status())
        .build();
```

Reply:
updated it.